open-apparel-registry/open-apparel-registry
# Implement Claim a Facility Dashboard

## Overview

This PR implements some MVP facility claim review dashboard pages which admin users can access to review and annotate FacilityClaims and also update statuses. There's a list page at

http://localhost:6543/dashboard/claims

and a detail page for each claim ID at

http://localhost:6543/dashboard/claims/claimID

On the details page, admin users can see data as depicted below and also update statuses and make notes.

On the API side, I added two migrations to store data about reviewing facility claims, added endpoints for approving, denying, and rejecting facility claims as well as adding new notes, set up the needed serializers, and added functionality to email facility claimants on status changes. (The emails for the latter are templates now but we can flesh them out in specific ways later).

I overloaded the facility claim notes field a little bit by using it to log details about status changes too. I did this specifically because I wanted the notes trail to be a log -- and not to have a separate status log.

Connects #517 

## Demo

![Screen Shot 2019-06-06 at 5 54 20 PM](https://user-images.githubusercontent.com/4165523/59069242-8c27f400-8884-11e9-9fc0-47058d84c20c.png)

![Screen Shot 2019-06-06 at 5 55 28 PM](https://user-images.githubusercontent.com/4165523/59069314-be395600-8884-11e9-9224-752fe2e4fa1c.png)

![Screen Shot 2019-06-06 at 5 56 12 PM](https://user-images.githubusercontent.com/4165523/59069323-c1ccdd00-8884-11e9-8da1-be91c71279e4.png)

![Screen Shot 2019-06-06 at 5 56 19 PM](https://user-images.githubusercontent.com/4165523/59069331-c72a2780-8884-11e9-8067-413d6e1cf60f.png)

![Screen Shot 2019-06-06 at 5 56 35 PM](https://user-images.githubusercontent.com/4165523/59069337-cb564500-8884-11e9-9d37-98138ee04cfe.png)

## Testing Instructions

- get this branch then `resetdb`, `processfixtures` and `createsuperuser`
- `./scripts/server`
- `./scripts/manage createsuperuser` then sign in to the Django Admin as your super user an switch on `claim_a_facility`
- register for a new account in the client, then verify it
- submit a couple of claim a facility forms, making the test submitted in each distinct enough to tell them apart
- sign out, then sign back in as your super user and visit http://localhost:6543/dashboard/claims and verify you see the facility claims table
    - clicking a one of the obvious links here should route you either to user profile or facility details pages
    - clicking a row *not* on a link should take you to the details page
- refresh a details page and verify that it loads the details
- try out adding some notes and verify that it works as expected
- try out approving and denying and verify that that works too
- try out revoking an approved facility and verify that that also works

## Checklist
